File: Src/CSharpier.Rider/src/main/java/com/intellij/csharpierrider/CSharpierProcessSingleFile.java
package com.intellij.csharpierrider;

import com.esotericsoftware.minlog.Log;
import com.intellij.openapi.diagnostic.Logger;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStream;

public class CSharpierProcessSingleFile implements ICSharpierProcess {
    Logger LOG = Logger.getInstance(CSharpierProcessSingleFile.class);
    String csharpierPath;

    public CSharpierProcessSingleFile(String csharpierPath) {
        this.csharpierPath = csharpierPath;
    }

    @Override
    public String formatFile(String content, String fileName) {
        try {
            ProcessBuilder processBuilder = new ProcessBuilder("dotnet", csharpierPath, "--write-stdout");
            processBuilder.redirectErrorStream(true);
            Process process = processBuilder.start();

            OutputStream stdin = process.getOutputStream();
            BufferedReader stdOut = new BufferedReader(new InputStreamReader(process.getInputStream()));

            stdin.write(content.getBytes());
            stdin.close();

            StringBuilder output = new StringBuilder();

            var nextCharacter = stdOut.read();
            while (nextCharacter != -1) {
                LOG.info("Got Output " + nextCharacter);
                output.append((char)nextCharacter);
                nextCharacter = stdOut.read();
            }

            String result = output.toString();

            if (process.exitValue() == 0 && !result.contains("Failed to compile so was not formatted.")) {
                return result;
            }
            else {
                Log.error(result);
            }
        } catch (Exception e) {
            LOG.error("error", e);
        }

        return "";
    }
}

File: Src/CSharpier.VisualStudio/CSharpier.VisualStudioShared/CSharpierProcessSingleFile.cs
using System.Diagnostics;
using System.Text;

namespace CSharpier.VisualStudio
{
    public class CSharpierProcessSingleFile : ICSharpierProcess
    {
        private readonly string csharpierPath;
        private readonly Logger logger;

        public string Version { get; }
        public bool ProcessFailedToStart => false;

        public CSharpierProcessSingleFile(string csharpierPath, string version, Logger logger)
        {
            this.csharpierPath = csharpierPath;
            this.logger = logger;
            this.Version = version;
        }

        public string FormatFile(string content, string fileName)
        {
            var output = new StringBuilder();
            var errorOutput = new StringBuilder();

            var processStartInfo = new ProcessStartInfo(this.csharpierPath)
            {
                RedirectStandardInput = true,
                RedirectStandardOutput = true,
                RedirectStandardError = true,
                UseShellExecute = false,
                CreateNoWindow = true,
            };
            processStartInfo.EnvironmentVariables["DOTNET_NOLOGO"] = "1";

            using var process = new Process { StartInfo = processStartInfo };
            process.Start();

            process.StandardInput.Write(content);
            process.StandardInput.Close();

            output.Append(process.StandardOutput.ReadToEnd());
            errorOutput.Append(process.StandardError.ReadToEnd());

            process.WaitForExit();

            var result = output.ToString();
            if (
                process.ExitCode == 0
                && !result.Contains("Failed to compile so was not formatted.")
            )
            {
                return result;
            }

            return string.Empty;
        }

        public void Dispose() { }
    }
}
